pilot/pkg/model/telemetry.go

```diff
@@ -462,41 +485,48 @@ func mergeLogs(logs []*tpb.AccessLogging, mesh *meshconfig.MeshConfig, mode tpb.
 	var loggingFilter *tpb.AccessLogging_Filter
 	providerNames := mesh.GetDefaultProviders().GetAccessLogging()
 	for _, m := range logs {
-		names := getProviderNames(m.Providers)
-		if len(names) > 0 {
-			providerNames = names
+		names := sets.New()
+		for _, p := range m.Logging {
+			subProviders := getProviderNames(p.Providers)
+			names.InsertAll(subProviders...)
+
+			if p.Filter != nil {
+				loggingFilter = p.Filter
```

Comment:
this seems like we just pick the last one, shouldn't it be per-block?

Reply:
`Filter` is hard to merged, I have no good idea on it from the begin, so I just keep as before.

Reply:
Shouldn't it not be merged at all?  Right now its not very safe -- we may try to enabl a provider for only some rare events, but then the filter is ignored and we send all logs?